src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -1212,6 +1212,14 @@ private void Dispose(bool disposing)
                     {
                         this.outputContext.Dispose();
                     }
+                    else if (this.settings.EnableMessageStreamDisposal)
+                    {
+                        Stream stream = this.message.GetStream();

```

Comment:
Do we need to always call GetStream? Instead, should we keep a reference to it in the ODataMessage and dispose of that if it is not null and call GetStream as part of else. 
 else if (this.settings.EnableMessageStreamDisposal && this.message.InternalStream != null)
                    {
                        this.message.InternalStream.dispose();
                    }
else if (EnableMessageStreamDisposal)
{
GetStream()
.Dispose();
}